### ISSUE TITLE
Prosthetic Limb Issues

### ISSUE THREAD

**Issue by Acrepe** [CleverRaven/Cataclysm-DDA#69856]
### Describe the bug

Current implementation of Prosthetic Limbs results in some strangeness; 
1: Prosthetic limbs "rub uncomfortably" against your hands and feet. The No Arm/Leg mutations, and the limbs themselves, should both prevent hands/feet from existing separately from the limbs in question.
2: Prosthetic limbs heal over time, instead of requiring repairs.
3: Prosthetic limbs can receive "deep bites", and presumably can become infected as well
4: Prosthetic limbs count as "rigid armor" on the location in question. This prevents wearing hard armor over them. 

### Attach save file

N/A

### Steps to reproduce

1: Spawn a character with one of the amputee starts
2: Suffer constant pain and speed reductions from your prosthetics rubbing against your nonexistent feet/hands
3: Receive damage
4: Watch as your plastic and metal limbs miraculously heal
5: Attempt to put on rigid armor. Fail.

### Expected behavior

Plastic/metal/wood should not be self-healing without supertech/blob magic, nor should it be impossible to layer armor over them. As well, No Arm/Prosthetic Arm and No Leg/Prosthetic Leg should not allow you to have floating flesh hands/feet that can be made uncomfortable by your own artificial limbs.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: 384d72f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Aftershock [aftershock],
    Mind Over Matter [mindovermatter]
]

### Additional context

It might be nice if prosthetics could be modified in some manner with late game tech, or have armor-plating inserts. The Exodii in particular come to mind.

**Comment by Willzadl** [comment]
Yep, constant speed reduction from chafing, though I didn't have any pain from the limb.

Prosthetic limbs do also heal on their own, seemingly also affected by bandaging them, acting as if they were a normal limb in terms of health.

I was also unable to put on rigid armour.
![Screenshot_2496](https://github.com/CleverRaven/Cataclysm-DDA/assets/152449730/bb893c78-9215-4e92-a614-2de86e9b5616)
![Screenshot_2497](https://github.com/CleverRaven/Cataclysm-DDA/assets/152449730/bc5dc22c-9f2e-4b3f-91cd-259d60840c92)

**Comment by MNG-cataclysm** [comment]
#69888 should fix issues with rigid armor and constant rubbing, knocking those two issues out. Limbs healing and the deep bites issues are still left.

**Comment by Maleclypse** [comment]
So am I understanding that the deep bites issue will need a new json field probably `"immune_to_effect": [ array ]`  so that various limbs can be immune to various effects?

**Comment by MNG-cataclysm** [comment]
> So am I understanding that the deep bites issue will need a new json field probably `"immune_to_effect": [ array ]` so that various limbs can be immune to various effects?

I might have a workaround for it, I’ll have to test. If that doesn’t work, then a new limb entry would be the best choice.

**Comment by Acrepe** [comment]
> #69888 should fix issues with rigid armor and constant rubbing, knocking those two issues out. Limbs healing and the deep bites issues are still left.

While I did fix the rubbing, it still seems to conflict with rigid armor
